### PR TITLE
Standardize common configuration telemetry

### PR DESCRIPTION
**Pull request by ygree**
# What Does This Do

- Collects null config defaults
- Serializes maps as string, empty maps as empty strings
- Standardizes common settings naming
  - Converts to snake case when reported with the telemetry
  - Introduces `logs_injection_enabled` with the `logs_injection` alias
  - Introduces `trace_tags` with the `tags` alias
- Standardize common settings values?
  - All setting values converted to strings except for `null`
    - Empty maps and lists reported as empty strings
  - Defaults the sampling rate to `null` (JUST AS IT WAS), which means sample everything, but is not exactly the same as if it's explicitly set to `1.0`
  - Allows reordering of map setting keys for some map settings (JUST AS IT WAS)
  - Allows lowercase keys for some merged map settings from different sources (JUST AS IT WAS)

# Motivation

In-app APM Library Configuration Reporting

# Additional Notes

Jira ticket: [APMJAVA-1217]

- [x] https://github.com/DataDog/system-tests/pull/2161 - to be merged with a fix for currently failing `test_app_started_client_configuration` 

[Corresponding Parametric Tests](https://github.com/DataDog/system-tests/blob/main/tests/parametric/test_telemetry.py)
[Similar Python Tracer change](https://github.com/DataDog/dd-trace-py/pull/7861)

[APMJAVA-1217]: https://datadoghq.atlassian.net/browse/APMJAVA-1217?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by PerfectSlayer** [comment]
Hey @ygree 👋 
According to the PR description, there is still some stuff to standardize.
Is the PR ready to review or should we wait for them first?

**Comment by ygree** [comment]
> Hey @ygree 👋 According to the PR description, there is still some stuff to standardize. Is the PR ready to review or should we wait for them first?

@PerfectSlayer We have discussed and I have resolved all TODOs

**Review by PerfectSlayer — APPROVED** [review]
Looks good to me but you might want another pair of eyes.
I got few minor questions but I am concerned about the default sampling rate change. Is it a breaking change or something worth notifying the customer on the release notes?

File: internal-api/src/main/java/datadog/trace/api/ConfigSetting.java [hunk 1]
```diff
@@ -14,16 +16,84 @@ public final class ConfigSetting {
...
+        result.append(',');
+      }
+      result.append(entry.getKey());
+      result.append(':');
+      result.append(entry.getValue());
+    }
+    return result.toString();
+  }
+
+  private static String renderIterable(Iterable iterable) {
```
[the remark this answers is not in the document]
Comment (mcculls) [reply internal-api/src/main/java/datadog/trace/api/ConfigSetting.java:86]: if we plan to escape delimiters in list values (like with map) then we'll want to keep this function

**Comment by ygree** [comment]
> Looks good to me but you might want another pair of eyes. I got few minor questions but I am concerned about the default sampling rate change. Is it a breaking change or something worth notifying the customer on the release notes?

We end up agreeing to leave the default sample rate untouched. I've changed the PR description to explicitly state this.

File: internal-api/src/main/java/datadog/trace/api/ConfigSetting.java [hunk 2]
```diff
@@ -14,16 +16,84 @@ public final class ConfigSetting {
...
+
+  private static String renderMap(Map<Object, Object> merged) {
+    StringBuilder result = new StringBuilder();
+    for (Map.Entry entry : merged.entrySet()) {
+      if (result.length() > 0) {
+        result.append(',');
+      }
+      result.append(entry.getKey());
+      result.append(':');
+      result.append(entry.getValue());
```
[the remark this answers is not in the document]
Comment (ygree) [reply internal-api/src/main/java/datadog/trace/api/ConfigSetting.java:81]: No, we shouldn't. Keys and values are taken from the parsed settings, which don't support escaping. See `ConfigConverter.loadMap` or `ConfigConverter.parseList`.

File: internal-api/src/main/java/datadog/trace/api/ConfigSetting.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ygree) [reply internal-api/src/main/java/datadog/trace/api/ConfigSetting.java]: > Can this method be replaced by JDK `String.join(",", iterable)`?

No, `String.join` accepts only `Iterable<? extends CharSequence>`

**Pull request merged**